R/fit_model.R

```diff
@@ -0,0 +1,30 @@
+fit_model <- function(data, formula, user_supplied_args) {
+  UseMethod("fit_model")
+}
+
+#' @export
+fit_model.RtGam_gam <- function(data, formula, user_supplied_args) {
+  default_args <- list(
+    formula = formula,
+    data = data,
+    family = "nb",
+    method = "REML"
+  )
+  args <- utils::modifyList(default_args, user_supplied_args)
+
+  do.call(mgcv::gam, args)
+}
+
+#' @export
+fit_model.RtGam_bam <- function(data, formula, user_supplied_args) {
```

Comment:
if you did do this with ... then potentially you could just set your defaults as actual defaults to args? Have you tested this?

Reply:
i.e avoid the modifyList etc.

Reply:
So I didn't think the methods would be private which was the main motivation but its also a very slight taste based thing to not have internal lists defining settings if it can be avoided.

Reply:
https://github.com/CDCgov/cfa-gam-rt/pull/20/commits/652ff5b29c58aa441e9877ac03c3b0c28034861e

Reply:
See repeat comment. Does using do.call on fit_model not work?